src/solidity/stateDecoder.js

```diff
@@ -0,0 +1,233 @@
+'use strict'
+var BN = require('ethereumjs-util').BN
+var utileth = require('ethereumjs-util')
+var varUtil = require('./variable')
+
+module.exports = {
+  decodeArray: function (type, storage, location) {
+    return this.decodeArrayItems(type, storage, 0, location)
+  },
+
+  decodeArrayItems: function (type, storage, depth, location) {
```

Comment:
Note that the encoding of arrays in memory is different from the encoding in calldata and again different in storage.